### DISCUSSION TITLE
Driver ignoring preferred scales for multilevel sensors

### DISCUSSION THREAD

**Discussion by spudwebb** [zwave-js/zwave-js#7284] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

In ZWave JS UI, I have preferred scales set to "temperature: Fahrenheit (°F)", if I refresh an "Air temperature" multilevel sensor I always get a value in Celsius.
In the logs I see:

`2024-10-15T16:17:35.933Z CNTRLR   [Node 010] Preferred scale "1" for sensor type 1 not found, using the first supported scale 0`

I believe the problem is in the following lines in `getPreferredSensorScale`. As `Object.entries()` always return keys as string, when the preferred scale is found, `preferred `is always set to the key as a string and then the second if is true as if the look up had failed:

https://github.com/zwave-js/node-zwave-js/blob/70f3550eb4113d0f553026b51d6cd2405513e43e/packages/cc/src/cc/MultilevelSensorCC.ts#L161-L181

### Software versions

Driver (node-zwave-js): 13.9.0

Z-Wave JS UI: 9.23.0

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-10-15.log](https://github.com/user-attachments/files/17381788/zwavejs_2024-10-15.log)

**Comment by kpine** [comment]
Looks like #7283 is reporting the same.

**Comment by AlCalzone** [comment, marked as answer]
Good catch! Escalated to an issue https://github.com/zwave-js/node-zwave-js/issues/7285
Fix will come ASAP.
